lib/sharing/ConsumeSharedRuntimeModule.js

```diff
@@ -166,10 +165,15 @@ class ConsumeSharedRuntimeModule extends RuntimeModule {
 					"throw new Error(getInvalidVersionMessage(scope, scopeName, key, requiredVersion));"
 				]
 			)};`,
+			`var warn = ${
+				this.compilation.options.output.ignoreBrowserWarnings
+					? "undefined"
+					: 'typeof console !== "undefined" && console.warn'
+			};`,
 			`var warnInvalidVersion = ${runtimeTemplate.basicFunction(
```

Comment:
To be hoesntly we still contain unused code in bundle, but we really don't need, just do more complex check and don't problem it when we have `ignoreBrowserWarnings: true`, although terser/swc remove them...

Reply:
I just pushed an extreme version of this. I think we should probably just set these `var warn` to empty functions. This isn't particularly maintainable although the test coverage seems really good. Is basically just to save a few lines of source in the browser that we had before anyways.

Reply:
To be clear, I was saying I don't really like this version. It's harder to maintain and seems very error prone. What is the downside of exposing `var warn = ` and making it an empty function when browser warnings are ignored and console.warn otherwise?

Reply:
@alexander-akait are you good with the current state or is there some medium ground? I am concerned about conditionally removing every time a warn is called in these two runtime modules as it's very error-prone and some future editor could easily introduce an inadvertent error.